internal/controller/cache.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package controller
+
+import (
+	"context"
+
+	inferencev1alpha1 "github.com/openshift/instaslice-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+	logr "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Cache avoids non determinism in the system which occurs when we play
+// catch game by being upto date with .status.podallocationresults. status
+// takes time to propagate and often causes controller to assign same slice
+// to multiple pods.
+
+func (r *InstasliceReconciler) rebuildAllocationCache(ctx context.Context) error {
```

Comment:
A lister automatically includes a cache. Perhaps we can use a lister.

https://github.com/kubernetes/client-go/blob/master/dynamic/dynamiclister/lister.go

Reply:
Thanks. Instead of doing an API server call, I will use Lister in the cache. the cache is needed to reuse the allocation decision made by the controller in capacity.go

Reply:
Reading the docs mgr.Client() is backed by a cache, turned out to be a simple fix, I don't think we need listers